include/aos/common/tools/hexconverter.hpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_HEXCONVERTER_HPP_
+#define AOS_HEXCONVERTER_HPP_
+
+#include "aos/common/tools/string.hpp"
+
+namespace aos {
+
+/**
+ * Hex converter
+ */
+class HexConverter {
+public:
+    /**
+     * Converts byte value to a hex string.
+     *
+     * @param val source byte value.
+     * @return StaticString<2>.
+     */
+    static StaticString<2> Convert(uint8_t val)
+    {
+        StaticString<2> dst;
+
+        constexpr char cDigits[] = "0123456789ABCDEF";
+
+        const auto low = val & 0xF;
+        const auto high = (val >> 4);
+
+        dst.PushBack(cDigits[high]);
+        dst.PushBack(cDigits[low]);
+
+        return dst;
+    }
+
+    /**
+     * Converts hex character to byte integer.
+     *
+     * @param ch source hex digit.
+     * @return RetWithError<uint8_t>.
+     */
+    static RetWithError<uint8_t> Convert(char ch)
+    {
+        if (ch >= '0' && ch <= '9') {
+            return static_cast<uint8_t>(ch - '0');
+        }
+
+        if (ch >= 'a' && ch <= 'f') {
+            return static_cast<uint8_t>(ch - 'a' + 10);
+        }
+
+        if (ch >= 'A' && ch <= 'F') {
+            return static_cast<uint8_t>(ch - 'A' + 10);
+        }
+
+        return {static_cast<uint8_t>(255), ErrorEnum::eInvalidArgument};
+    }
+
+    /**
+     * Converts byte array to a hex string.
+     *
+     * @param src source byte array.
+     * @param dst result string.
+     * @return Error.
+     */
+    static Error Convert(const Array<uint8_t>& src, String& dst)
+    {
+        if (dst.MaxSize() < src.Size() * 2) {
+            return ErrorEnum::eNoMemory;
+        }
+
+        dst.Clear();
+
+        for (const auto val : src) {
+            const auto digits = Convert(val);
+
+            dst.PushBack(digits[0]);
+            dst.PushBack(digits[1]);
+        }
+
+        return ErrorEnum::eNone;
+    }
+
+    /**
+     * Converts hex string to byte array.
+     *
+     * @param src source hex string.
+     * @param dst result byte array.
+     * @return Error.
+     */
+    static Error Convert(const String& src, Array<uint8_t>& dst)
```

Comment:
Remove hexconverter at all, use String::ToByteArr and Convert.

Reply:
Hex converter looks better because Array<byte> <=> String convertion depends on the context. 
And sometimes it is implemented as hex string convertion, sometimes with reinterepret_cast. HexConverter removes this ambiguity. Another reason for HexConverter is that uuid implementation logic looks easier with extra methods to convert char<=> byte, then reusing Array<byte> <=> String converters.

Reply:
ok, I moved implementation back to String class.